lib/erl_docgen/priv/bin/chunk.escript

```diff
@@ -0,0 +1,738 @@
+      end || N = {name,_,_} <- Dom].
+
+transform_seealso(_S = {seealso,_Attr,_Content}) ->
+    _Content.
+
+to_chunk(Dom, Source, Module, AST) ->
+    [{module,MAttr,Mcontent}] = Dom,
+
+    ModuleDocs = lists:flatmap(
+                   fun({p,_,Content}) ->
+                           Content;
+                      ({_,_,_}) ->
+                           []
+                   end, Mcontent),
+
+    TypeMeta = add_types(AST, maps:from_list([{source,Source}|MAttr])),
+
+    TypeMap = maps:get(types, TypeMeta, []),
+
+    Anno = erl_anno:set_file(atom_to_list(Module)++".erl",erl_anno:new(0)),
+
+    Types = lists:flatten([Types || {datatypes,[],Types} <- Mcontent]),
+
+    TypeEntries =
+        lists:map(
+          fun({datatype,Attr,Descr}) ->
+                  TypeName = func_to_atom(proplists:get_value(name,Attr)),
+                  TypeArity = case proplists:get_value(n_vars,Attr) of
+                                  undefined ->
+                                      find_type_arity(TypeName, TypeMap);
+                                  Arity ->
+                                      list_to_integer(Arity)
+                              end,
+                  TypeArgs = lists:join(",",[lists:concat(["Arg",I]) || I <- lists:seq(1,TypeArity)]),
+                  PlaceholderSig = io_lib:format("-type ~p(~s) :: term().",[TypeName,TypeArgs]),
+                  TypeSignature = proplists:get_value(
+                                    signature,Attr,[iolist_to_binary(PlaceholderSig)]),
+                  MetaSig =
+                      case maps:get({TypeName, TypeArity}, TypeMap, undefined) of
+                          undefined ->
+                              #{};
+                          Sig ->
+                              #{ signature => [Sig] }
+                      end,
+                  docs_v1_entry(type, Anno, TypeName, TypeArity, TypeSignature, MetaSig, Descr)
+          end, Types),
+
+    Functions = lists:flatten([Functions || {functions,[],Functions} <- Mcontent]),
+
+    FuncEntrys =
+        lists:flatmap(
+          fun({function,Attr,Fdoc}) ->
+                  case func_to_atom(proplists:get_value(name,Attr)) of
+                      callback ->
+                          [];
+                      Name ->
+                          Arity = proplists:get_value(arity,Attr),
+                          Signature = proplists:get_value(signature,Attr),
+                          FMeta = proplists:get_value(meta,Attr),
+                          MetaWSpec = add_spec(AST,FMeta),
+                          [docs_v1_entry(function, Anno, Name, Arity, Signature, MetaWSpec, Fdoc)]
+                  end
+          end, Functions),
+
+    docs_v1(ModuleDocs, Anno, TypeMeta, FuncEntrys ++ TypeEntries).
+
+docs_v1(DocContents, Anno, Metadata, Docs) ->
+    #docs_v1{ anno = Anno,
+              module_doc = #{<<"en">> => term_to_binary(shell_docs:normalize(DocContents))},
```

Comment:
We had discussions about this and we were considering updating EEP 48 to allow any term to be given here, so we don't do a double `term_to_binary`. What are your final thoughts? I can send a PR to EEP 48 later today if we want to lift this restriction. We already need to update it to mention `edit_url`.

Reply:
Yes, I think that would be a nice change.

Reply:
PR sent: https://github.com/erlang/eep/pull/9

Reply:
thanks!